cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -650,12 +634,14 @@ func main() {
 			}
 		}()
 
+		cancel()
```

Comment:
If this is canceled before proxyClient.Shutdown, won't it end all of the connections that you've passed it to?

Reply:
Yes it would. I suppose we should give clients `termTimeout` to close themselves, and then cancel the context after that warning. I'll make this run on function exit.